compiler/src/dotty/tools/dotc/util/Signatures.scala

```diff
@@ -84,10 +85,54 @@ object Signatures {
 
     (paramIndex, alternativeIndex, alternatives)
 
+  private def unapplyCallInfo(
+    span: Span,
+    fun: tpd.Tree,
+    patterns: List[tpd.Tree]
+  )(using Context): (Int, Int, List[SingleDenotation]) =
+    val patternPosition = patterns.indexWhere(_.span.contains(span))
+    val activeParameter = extractParamTypess(fun.tpe.finalResultType.widen).headOption.map { params =>
```

Comment:
Possible performance optimization as this is computed twice. We need to find number of parameters we can unapply to but i can't pass it without changing the API